### PR TITLE
Deleting more objects

### PR DESCRIPTION
**Pull request by paigerube14**
Moving changes from [pr](https://github.com/redhat-chaos/krkn-lib/pull/44) to branch in redhat-chaos org to be able to run with secrets

### CODE REVIEW

**Review by chaitanyaenr — APPROVED** [review]
LGTM!

**Review by tsebastiani — CHANGES_REQUESTED** [review]
Hi Paige,
Would be great if you could:

- add  unit tests to the methods that you added otherwise the coverage will drop ( I know that sometimes may feel odd to write tests to small methods with an obvious outcome, but we've to try to keep the coverage high)
- add types to the method parameters as well to align with the lib convention
- add some pydoc to the methods so the doc website will be populated properly.

added some comments in the code as well

Thanks a lot!

File: src/krkn_lib/k8s/krkn_kubernetes.py [hunk 1]
```diff
@@ -434,6 +435,101 @@ def list_pods(
...
+    def get_deployment_ns(self, namespace) -> list[str]:
+        """
+        Return a list of tuples containing pod name [0] and namespace [1]
+
+        :param label_selector: filter by label_selector
+            (optional default `None`)
+        :return: list of tuples pod,namespace
+        """
+        deployments = []
+        ret = self.apps_api.list_namespaced_deployment(namespace, pretty=True)
```
Comment (tsebastiani) [inline src/krkn_lib/k8s/krkn_kubernetes.py:461]: not catching exception

**Comment by paigerube14** [comment]
@tsebastiani @chaitanyaenr can you ptal, I have updated with pydoc and tests

File: pyproject.toml [hunk 2]
```diff
@@ -10,13 +10,14 @@ homepage = "https://github.com/redhat-chaos/krkn"
 
 [tool.poetry.dependencies]
 python = "^3.9"
-kubernetes =">=26.1,<29.0"
+kubernetes ="^28.1.0"
```
Comment (chaitanyaenr) [inline pyproject.toml:13]: Hitting the following during the install:
```
The conflict is caused by:
    The user requested kubernetes
    krkn-lib 0.0.0 depends on kubernetes<29.0.0 and >=28.1.0
    arcaflow-plugin-kill-pod 0.0.0 depends on kubernetes<27.0.0 and >=26.1.0
```

**Review by tsebastiani — APPROVED** [review]

**Pull request merged**